src/main/scala/dsptools/numbers/algebra_types/Signed.scala

```diff
@@ -42,6 +43,11 @@ object Signed {
 
 private class OrderedRingIsSigned[A <: Data](implicit o: Order[A], r: Ring[A]) extends Signed[A] {
   def signum(a: A): ComparisonBundle = o.compare(a, r.zero)
-  def abs(a: A): A = Mux(signum(a).lt, r.negate(a), a)
+  def abs(a: A): A = {
+    Mux(signum(a).lt, r.negate(a), a)  //TODO: should this be negateContext
+  }
+  def context_abs(a: A): A = {
```

Comment:
For operations like + and -, don't you sometimes pipeline every three adds, or something like that? Perhaps we simply shouldn't have a definition here.